src/reducers/arcReducer.ts

```diff
@@ -229,9 +231,10 @@ export const initialState: IArcState = {
 };
 
 export const closingTime = (proposal: IProposalState) => {
-  const { state, boostedTime, submittedTime, preBoostedVotePeriodLimit, boostedVotePeriodLimit, executionTime } = proposal;
+  const { state, boostedTime, submittedTime, preBoostedVotePeriodLimit, boostedVotePeriodLimit, quietPeriod, executionTime } = proposal;
+
   const start = boostedTime ? boostedTime : submittedTime;
-  const duration = boostedTime ? boostedVotePeriodLimit : preBoostedVotePeriodLimit;
+  const duration = boostedTime ? (state === ProposalStates.QuietEndingPeriod ? quietPeriod : boostedVotePeriodLimit) : preBoostedVotePeriodLimit;
```

Comment:
yeah i don't think this is right, i think my fix in a9136f3e9b64eba68d07b7440e93a0c0e336e57c handles it, because Arc actually updates the boostedTime and the boostedVotePeriodLimit when it hits the quiet ending period

Reply:
`boostedTime` time is updated to be the time when the decision was flipped. `boostedVotePeriodLimit` is a DAO param though and does not change. Once the the proposal is in quiet mode the calculation should be `boostedTime + quietPeriod`.

Reply:
proposal.currentBoostedVotePeriodLimit is the one we care about, and Arc does update it

Reply:
correct, was confused by the naming because `boostedVotePeriodLimit` is a DAO param in Arc.

Reply:
yeah, that would be a good thing to change